src/include/types.h

```diff
@@ -541,5 +541,33 @@ WRITE_CLASS_ENCODER(errorcode32_t)
 WRITE_EQ_OPERATORS_1(errorcode32_t, code)
 WRITE_CMP_OPERATORS_1(errorcode32_t, code)
 
+struct sha1_fp_t {
```

Comment:
nit: can we call this sha1_digest_t (fp == fingerprint i assume, but digest is clearer)